workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -352,6 +355,62 @@ def _restore_deployment_envs(self, postgres):
 
         self._log_final_information(deps_with_failed_plugins)
 
+    def _restore_inter_deployment_dependencies(self):
+        if (self._snapshot_version < V_4_6_0 or
+                self._snapshot_version > V_5_0_5):
+            return
+        ctx.logger.info('Restoring inter deployment dependencies')
+        update_service_composition = (self._snapshot_version == V_5_0_5)
+
+        deployment_contexts = utils.get_dep_contexts(self._snapshot_version)
+        deployments_queue = queue.Queue()
+        failed_deployments_queue = queue.Queue()
+        for tenant_name, deployments in deployment_contexts:
+            for dep_id in deployments:
+                deployments_queue.put((tenant_name, dep_id))
+
+        wf_context = current_workflow_ctx.get_ctx()
+        context_params = current_workflow_ctx.get_parameters()
+
+        def _create_inter_deployment_dependencies():
```

Comment:
i would prefer this function to take both queues as arguments, instead of by closure (but it would also need wf_context and params then)

the reason is that this is going to run as a thread, and with threads, you want to really be sure that you don't have unsynchronized (unlocked) access to random things. If this function didn't take anything by closure, but instead took everything by arguments, then it would even be possible to move it outside, to module level - and then it would be very obvious that it doesn't access anything it shouldn't

accessing things by closure like this, just makes me look at this function very closely to examine if it doesn't touch anything that it isn't supposed to

Reply:
Makes a lot of sense, changed it